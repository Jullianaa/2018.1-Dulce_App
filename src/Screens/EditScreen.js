import React from 'react';
<<<<<<< HEAD
import {ScrollView, View, TextInput, Text, TouchableHighlight, Image} from 'react-native';
=======
import {View, Text} from 'react-native';
>>>>>>> 5beeb69... #194 arrumando parte das telas
import AGRButton from '../Components/AGRButton';
import AGRInput from '../Components/AGRInput';
import SmallLogo from '../Components/SmallLogo';
import ValidationComponent from 'react-native-form-validator';
import store from '../Reducers/store';
<<<<<<< HEAD
<<<<<<< HEAD
import axios from 'axios';
import { Container, Header, Content, Spinner } from 'native-base';
=======
>>>>>>> 85748c1... #194 dando comando eslint --fix
=======
import axios from 'axios';
>>>>>>> 5beeb69... #194 arrumando parte das telas

const styles = {
  container: {
    flex: 1,
    flexDirection: 'column',
    padding: 15,
    backgroundColor: '#FFF',
    paddingHorizontal: 20
  },
  input: {
    height: 40,
    borderBottomWidth: 1,
    borderBottomColor: 'grey',
    marginBottom: 10,
    alignSelf: 'center',
    width: '80%',
    marginTop: 10
  },
  button: {
    alignItems: 'center',
    backgroundColor: 'lightgray',
    padding: 10,
    width: 140,
    height: 40,
    marginTop: 15,
    marginRight: 5,
    marginLeft: 10
  },
  alinhar: {
    flexDirection: 'row',
    alignContent: 'space-around'
  }
};
export default class EditScreen extends ValidationComponent {
  constructor(props) {
    super(props);
    this.state = {
      editable: true,
      collaborator: {},
      password: '',
      name: '',
      registration: '',
      hospital: '',
      sector:'',
      loading: true
    };
  }

  save() {
  axios.post('http://172.17.0.1:8080/user/edit/' + store.getState().currentUser.id , {

    password: this.state.password
  })
    .then((response) => {
      const token = response.data.token;
      this.props.navigation.navigate('list', {token: token});
    });
}
  _onPressButton() {
    if (this.validate({
      name: {required: true},
      registration: {numbers: true, required: true},
      hospital: {required: true},
      sector: {required: true},
      password: {minlength: 4, maxlength: 8, required: true}
    })) {this.save();}
  }
  save() {
    axios.post('http://172.17.0.1:8080/user/edit/' + store.getState().currentUser.id , {
      name: this.state.name,
      registration: this.state.registration,
      hospital: this.state.hospital,
      sector: this.state.sector,
      password: this.state.password
    })
      .then((response) => {
       this.props.navigation.navigate('list');
      })
  }
  componentDidMount(){
    this.setState({loading: true});

    const url = 'http://172.17.0.1:8080/user/view/' +  store.getState().currentUser.id ;


    axios.get(url,{

      headers: {
        'x-access-token': store.getState().currentUser.token
      }

    })

    .then((response) => {this.setState({collaborator: response.data,loading: false})
      this.setState({
        name: this.state.collaborator.name,
        registration: this.state.collaborator.registration,
        hospital: this.state.collaborator.hospital,
        sector: this.state.collaborator.sector
      });

    ;})
}
    }))
  { this.save()}


}
  render() {
    return this.state.loading ? (

      <View/>

        ) : (
        <View style={styles.container}>
        <SmallLogo/>
        <ScrollView>
        <AGRInput style={styles.input } value = {this.state.name}
          editable = {false}
          onChangeText={(text) => this.setState({
            name: text})}
        />
        {this.isFieldInError('name') && this.getErrorsInField('name').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
        <AGRInput style={styles.input} value = {this.state.registration}
          placeholder='Matricula'
          editable = {false}
          onChangeText={(text) => this.setState({
            registration: text})}
        />
        <AGRInput style={styles.input} value = {this.state.hospital}
          nameLabel='Hospital'
          editable = {this.state.editable}
          onChangeText={(text) => this.setState({
            name: text})}
        />
        {this.isFieldInError('hospital') && this.getErrorsInField('hospital').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
        <AGRInput style={styles.input} value = {this.state.sector}
          nameLabel='Setor'
          editable = {this.state.editable}
          onChangeText={(text) => this.setState({
            sector: text})}
        />
        {this.isFieldInError('sector') && this.getErrorsInField('sector').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
        <AGRInput style={styles.input}
          nameLabel='Editar senha'
          editable = {this.state.editable}
          secureTextEntry
          onChangeText={(text) => this.setState({
            password: text})}
        />
        {this.isFieldInError('password') && this.getErrorsInField('password').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
        <View style={styles.alinhar}>
          <AGRButton style={styles.button} text='Salvar' onPress={() => {this.save(); this._onPressButton();}}/>
        </View>
      );
  }
}
