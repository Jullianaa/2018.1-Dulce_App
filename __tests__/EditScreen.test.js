import 'react-native';
import React from 'react';
import EditScreen from '../src/Screens/EditScreen';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
  .create(<EditScreen />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
