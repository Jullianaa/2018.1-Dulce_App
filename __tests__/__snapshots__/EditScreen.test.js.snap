// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
      "flexDirection": "column",
      "padding": 15,
      "paddingHorizontal": 20,
    }
  }
>
  <View>
    <Image
      source={
        Object {
          "testUri": "../../../assets/img/logo.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "height": 100,
          "marginBottom": 20,
          "resizeMode": "contain",
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={Object {}}
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "left": 0,
            "opacity": 1,
            "paddingBottom": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 0,
            "top": 18,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#575757",
              "fontSize": 17,
              "left": 0,
              "paddingRight": 5,
              "top": 6,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 17,
              }
            }
            virtual={undefined}
          />
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#575757"
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "lineHeight": 24,
              "marginLeft": null,
              "paddingLeft": null,
              "paddingRight": 5,
              "top": 10,
            },
            Object {
              "alignSelf": "center",
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "height": 40,
              "marginBottom": 10,
              "marginTop": 10,
              "width": "80%",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value={undefined}
        virtual={undefined}
      />
    </View>
  </View>
  <View
    style={Object {}}
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "left": 0,
            "opacity": 1,
            "paddingBottom": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 0,
            "top": 18,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#575757",
              "fontSize": 17,
              "left": 0,
              "paddingRight": 5,
              "top": 6,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 17,
              }
            }
            virtual={undefined}
          />
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Matricula"
        placeholderTextColor="#575757"
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "lineHeight": 24,
              "marginLeft": null,
              "paddingLeft": null,
              "paddingRight": 5,
              "top": 10,
            },
            Object {
              "alignSelf": "center",
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "height": 40,
              "marginBottom": 10,
              "marginTop": 10,
              "width": "80%",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value={undefined}
        virtual={undefined}
      />
    </View>
  </View>
  <View
    style={Object {}}
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "left": 0,
            "opacity": 1,
            "paddingBottom": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 0,
            "top": 18,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#575757",
              "fontSize": 17,
              "left": 0,
              "paddingRight": 5,
              "top": 6,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 17,
              }
            }
            virtual={undefined}
          >
            Hospital
          </Text>
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        nameLabel="Hospital"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#575757"
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "lineHeight": 24,
              "marginLeft": null,
              "paddingLeft": null,
              "paddingRight": 5,
              "top": 10,
            },
            Object {
              "alignSelf": "center",
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "height": 40,
              "marginBottom": 10,
              "marginTop": 10,
              "width": "80%",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value={undefined}
        virtual={undefined}
      />
    </View>
  </View>
  <View
    style={Object {}}
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "left": 0,
            "opacity": 1,
            "paddingBottom": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 0,
            "top": 18,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#575757",
              "fontSize": 17,
              "left": 0,
              "paddingRight": 5,
              "top": 6,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 17,
              }
            }
            virtual={undefined}
          >
            setor
          </Text>
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        nameLabel="setor"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#575757"
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "lineHeight": 24,
              "marginLeft": null,
              "paddingLeft": null,
              "paddingRight": 5,
              "top": 10,
            },
            Object {
              "alignSelf": "center",
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "height": 40,
              "marginBottom": 10,
              "marginTop": 10,
              "width": "80%",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value={undefined}
        virtual={undefined}
      />
    </View>
  </View>
  <View
    style={Object {}}
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "left": 0,
            "opacity": 1,
            "paddingBottom": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 0,
            "top": 18,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#575757",
              "fontSize": 17,
              "left": 0,
              "paddingRight": 5,
              "top": 6,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 17,
              }
            }
            virtual={undefined}
          >
            Editar senha
          </Text>
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        nameLabel="Editar senha"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="#575757"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "lineHeight": 24,
              "marginLeft": null,
              "paddingLeft": null,
              "paddingRight": 5,
              "top": 10,
            },
            Object {
              "alignSelf": "center",
              "borderBottomColor": "grey",
              "borderBottomWidth": 1,
              "height": 40,
              "marginBottom": 10,
              "marginTop": 10,
              "width": "80%",
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
        virtual={undefined}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignContent": "space-around",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "height": 1334,
        }
      }
      virtual={undefined}
    >
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={
          Array [
            Object {
              "padding": undefined,
            },
            undefined,
          ]
        }
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutomaticScroll={true}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        resetScrollToCoords={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          }
        }
        viewIsInsideTabBar={false}
        virtual={undefined}
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "purple",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "elevation": 2,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "marginBottom": 0,
                "marginTop": 30,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingHorizontal": 50,
                "paddingTop": 6,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                }
              }
            >
              Salvar
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
